calculator_test.go

```diff
@@ -3,189 +3,427 @@ package calculator_test
+			want:        20,
+			errExpected: false,
+		},
+		{
+			name:        "Divide by 0 in slice",
+			a:           20,
+			b:           2,
+			nums:        []float64{2, 0},
+			want:        0,
 			errExpected: true,
 		},
 	}
 
-	for _, test := range successTestCases {
-		got, err := calculator.Divide(test.a, test.b)
+	for _, tc := range tcs {
+		got, err := calculator.Divide(tc.a, tc.b, tc.nums...)
 		errReceived := err != nil
-		if test.errExpected != errReceived{
-				t.Errorf("Test Divide (%f, %f) %s failed: Unexpected error received",
-					test.a, test.b, test.name)
-		} else {
-			if !test.errExpected && !almostEqual(test.want, got) {
-				t.Errorf("Test Divide (%f, %f) %s failed: want %f, got %f",
-					test.a, test.b, test.name, test.want, got)
-			}
+		if tc.errExpected != errReceived {
+			t.Errorf("%s: Divide (%f, %f, %v) : Unexpected error received",
+				tc.name, tc.a, tc.b, tc.nums)
+		}
+		if !tc.errExpected && !closeEnough(tc.want, got) {
+			t.Errorf("%s: Divide(%f, %f, %v) : want %f, got %f",
+				tc.name, tc.a, tc.b, tc.nums, tc.want, got)
 		}
+	}
+}
 
+func TestSquareRoot(t *testing.T) {
+	t.Parallel()
+	tcs := []struct {
+		name        string
+		a           float64
+		want        float64
+		errExpected bool
+	}{
+		{
+			name:        "Squareroot of positive number",
+			a:           16,
+			want:        4,
+			errExpected: false,
+		},
+		{
+			name:        "Squareroot of 0",
+			a:           0,
+			want:        0,
+			errExpected: false,
+		},
+		{
+			name:        "Squareroot of negative number",
+			a:           -1,
+			want:        0,
+			errExpected: true,
+		},
 	}
 
+	for _, tc := range tcs {
+		got, err := calculator.SquareRoot(tc.a)
+		errReceived := err != nil
+		if tc.errExpected != errReceived {
+			t.Errorf("%s: Squareroot(%f) : Unexpected error received",
+				tc.name, tc.a)
+		}
+		if !tc.errExpected && !closeEnough(tc.want, got) {
+			t.Errorf("%s: Squareroot(%f) : want %f, got %f",
+				tc.name, tc.a, tc.want, got)
+		}
+	}
+}
+
+func TestComputeString(t *testing.T) {
+	t.Parallel()
+	tcs := []struct {
+		name        string
+		input       string
+		want        float64
+		errExpected bool
+	}{
+		{
+			name:        "Addition of two numbers with no spaces",
+			input:       "2+2",
+			want:        4,
+			errExpected: false,
+		},
+		{
+			name:        "Subtraction with embedded space",
+			input:       "10     -     2",
+			want:        8,
+			errExpected: false,
+		},
+		{
+			name:        "Multiplication with fraction",
+			input:       "    4 *1.5",
+			want:        6,
+			errExpected: false,
+		},
+		{
+			name:        "Division of two fraction",
+			input:       "    4.4 /2.2",
+			want:        2.0,
+			errExpected: false,
+		},
+		{
+			name:        "Unsupported operator",
+			input:       "2%5",
+			want:        0,
+			errExpected: true,
+		},
+		{
+			name:        "Single operand",
+			input:       "2+",
+			want:        0,
+			errExpected: true,
+		},
+		{
+			name:        "More than 2 operands",
```

Comment:
Very good! You've got to work against yourself when designing good test cases: think about what's likely to break your code! That doesn't usually come naturally, since as engineers we like to think that it _will_ work, but it's important all the same.